modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/utils.es.js

```diff
@@ -32,12 +31,14 @@ export const createNewGroup = (items) => ({
 	items,
 });
 
+let uniqueIdCounter_ = 1;
+
 /**
  * Generates a unique group id.
  * @return {string} The unique id.
  */
 export function generateGroupId() {
-	return `${GROUP_ID_NAMESPACE}${getUid()}`;
+	return `${GROUP_ID_NAMESPACE}${uniqueIdCounter_++}`;
```

Comment:
Most of these change LGTM, but I have a doubt about this one.

I know that in most cases when we wanted uuid's we used the `uuid` package which is already in portal. 

An example can be seen in [`frontend-js-web/package.json`](https://github.com/liferay/liferay-portal/blob/b9a09a2d90eb15a0e4e307faef0e86b6107b6437/modules/apps/frontend-js/frontend-js-web/package.json#L19)

Reply:
Since we don't expose the `uuid` package from `frontend-js-web` or use it anywhere else, I think it should be fine to use this simple implementation here and might be overkill to make `uuid` package available everywhere or add another dependency to this module.

Reply:
@bryceosterhaus my bad, I didn't know we had to use something we were exposing from `frontend-js-web`.

We do use `uuid` in various places: [it's present in the `yarn.lock` file](https://github.com/liferay/liferay-portal/blob/769ee4eede8369fe97a188d8cadabee8222fad32/modules/yarn.lock#L15275-L15278) 
and from the `modules` directory if I run `yarn why uuid`, I get this output:

```sh
$ cd modules/
$ yarn why uuid
yarn why v1.22.10
[1/4] Why do we have the module "uuid"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "uuid@3.4.0"
info Reasons this module exists
   - "_project_#frontend-js-web" depends on it
   - Hoisted from "_project_#frontend-js-web#uuid"
   - Hoisted from "_project_#@liferay#npm-scripts#liferay-npm-bundler#insight#uuid"
   - Hoisted from "_project_#@liferay#npm-scripts#webpack-dev-server#webpack-log#uuid"
   - Hoisted from "_project_#@liferay#npm-scripts#webpack-dev-server#sockjs#uuid"
   - Hoisted from "_project_#frontend-theme-font-awesome-web#liferay-font-awesome#less#request#uuid"
info Disk size without dependencies: "92KB"
info Disk size with unique dependencies: "92KB"
info Disk size with transitive dependencies: "92KB"
info Number of shared dependencies: 0
Done in 1.39s.
```

I thought that the point of using yarn's workspace feature was to be able to have the same dependency declared in various (osgi) modules and installed in once place (`modules/node_modules`).

Going from metal's `getUid()` function to a simple "counter" seems like a step backwards (in the sense that it's less "robust"), but if you're sure that it's enough, then OK from me.

Reply:
> Going from metal's getUid() function to a simple "counter" seems like a step backwards (in the sense that it's less "robust"), but if you're sure that it's enough, then OK from me.

Metal's `getUid()` is a simple counter: https://github.com/metal/metal.js/blob/master/packages/metal/src/coreNamed.js#L163

Reply:
I just realized that but forgot to update my previous comment, because it's irrelevant.

Reply:
> I just realized that but forgot to update my previous comment, because it's irrelevant.

Why would you think it's irrelevant?

This is some of those things where we need to analyze what's going on. We usually will want to consume from `frontend-js-web` or a shared place to make sure we're not bloating our pages with the same libraries over and over again. At the same time, we need to analyze if the dependency brings enough value for its cost (or as @wincent likes to say, is the juice worth the squeeze?). Put all that together and we can figure out a path forward :)

Reply:
I'm coming late to the conversation but monotonically-increasing counters are fantastic (optimal, even) as unique ids if they only need to exist on the client. You only need the pull out the heavy-guns and use a UUID if the id is going to be shared across contexts (eg. sent to the server, persisted to the DB etc).

So we will need a shared UUID implementation for sure, _but_ we want to make sure we use it only where it's actually needed.